[WFLY-17110] Add Elytron WildFly Security Manager documentation

boris-unckel opened the pull request:
Fixes https://issues.redhat.com/browse/WFLY-17110

boris-unckel committed:
[WFLY-17110] Add Elytron WildFly Security Manager documentation

Fixes https://issues.redhat.com/browse/WFLY-17110

jasinner commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
```diff
@@ -0,0 +1,261 @@
+Running a JSM will not fully protect the server from malicious features of untrusted code. It does, however, offer another layer of protection which can help reduce the impact of serious security vulnerabilities, such as deserialization attacks. For example, most of the recent https://www.github.com/mbechler/marshalsec/blob/master/marshalsec.pdf?raw=true[attacks against Jackson Databind] rely on  ...
+
+==== JavaSE 17 Deprecate the Security Manager for Removal
+Unfortunately the community decided to deprecate the Java Security Manager with Java 17. The full details are described in https://openjdk.org/jeps/411[JEP 411: Deprecate the Security Manager for Removal]. WildFly {wildflyversion} still tests and actively supports the Security Manager. Especially for enterprise java security there is no better tool in place or development. WildFly developers stil ...
```
```suggestion
Unfortunately the community decided to deprecate the Java Security Manager with Java 17. The full details are described in https://openjdk.org/jeps/411[JEP 411: Deprecate the Security Manager for Removal]. WildFly {wildflyversion} still tests and actively supports the Security Manager. Used in conjunction with other tools such as [Serialization Filtering](https://docs.oracle.com/en/java/javase/16/core/serialization-filtering1.html#GUID-3ECB288D-E5BD-4412-892F-E9BB11D4C98A) the JSM is still a good defense in depth measure.
```

jasinner reviewed (approved):
Looks good to me. I added a few suggested improvements which I think make more sense in light of the references used in the text. The references all look good, thanks!

boris-unckel commented:
@jasinner 
> Looks good to me. I added a few suggested improvements which I think make more sense in light of the references used in the text. The references all look good, thanks!

Thanks for your feedback. I have fixed all the issues.

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
```diff
@@ -0,0 +1,327 @@
+* Support of minimum and maximum permissions per server instance.
+* Support of a "log-only" mode to ease development and testing of permissions.
+* Output of debug information to detect missing permissions.
+* Output of trace information to analyse special situations with {oracle-javadoc}/java.base/java/security/ProtectionDomain.html[protection domains], classloaders, and {oracle-javadoc}/java.base/java/security/CodeSource.html[codesources].
```
@boris-unckel  Just a minor, the links do not render properly

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
Just a minor, I would not use the word `unfortunately` in documentation as it is just an opinion and would depend on the context? And the links do not render well here

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
Debug is enabled via the following CLI commands

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
We should use link to the documentation and not link to the proposal if available

boris-unckel replied:
Thanks, found it - fixed.

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
@boris-unckel  I know this is taken from zulip. But when reading this documentation it is not clear what "our case" mean

boris-unckel replied:
```The WildFly modules case is``` ...

Skyllarr commented:
> @Skyllarr Thanks for your time to review and the valuable feedback. I have fixed the issues. Regarding the links: What are you using for review? Github won't work. In Intellij check `File | Settings | Languages & Frameworks | AsciiDoc` then add "Attributes" from here https://github.com/wildfly/wildfly/blob/main/docs/pom.xml#L65-L70 obviously with meaningful values. If you want to check the generated .html, simply cd [WFLYSRC]/docs `mvn clean install` (I don't remember exactly, maybe JAVA_HOME must be set before).

Thank you! I will check

boris-unckel commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
> PrivilegedAction is an interface which allows a `Runnable` class to be passed in and also has a return type.

Fixed (not yet pushed)

> Also in this sentence: `If this was added later they could have just used the functional interfaces. ` who is they?

If this was added later it could have been implemented with the functional interfaces.

boris-unckel commented:
@Skyllarr Thanks for the additional feedback. I have fixed all mentioned issues. If it's completed it would be great to get the approval for the next process steps, please.

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
@boris-unckel Thank you for the updates! 

Just a minor, but would you be able to change slighty the style of these last chapters? Since this was taken from zulip it reads more like a conversation instead of documentation. WDYT?

boris-unckel replied:
Ok, I assume you mean to switch from "our"/"we"... to a passive form. I was afraid my English leads to worse result but you're right - it must be a documentation style wording.

Skyllarr replied:
Yes, documentation style wording would be best. Also removing the question/answer style would be good. So taht the documentation is compact. Thank you!

boris-unckel replied:
@Skyllarr Done, hopefully correct

Skyllarr replied:
@boris-unckel Great, thank you! It reads much better now! I just added some remaining suggestions

Skyllarr commented on docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc:
```diff
@@ -0,0 +1,324 @@
+== Overview
+
+=== General introduction
+The Elytron WildFly Java Security Manager ist build on top of and within the https://docs.oracle.com/en/java/javase/11/security/java-se-platform-security-architecture.html[Java SE Platform Security Architecture].
```
s/ist/is

boris-unckel commented:
@Skyllarr Thanks for finding this. I have fixed all the issues. Please check

skyllarr reviewed (approved):
@boris-unckel  Thank you for the updates

bstansberry commented:
Thanks @boris-unckel @jasinner and @Skyllarr!
